### PR TITLE
overhaul tracer with events and websocket tunnels

### PR DESCRIPTION
**Pull request by adtac**

### CODE REVIEW

File: cmd/run/socket/proxy.go [hunk 1]
```diff
@@ -230,24 +241,98 @@ func (p *proxy) proxyHTTP(cli, srv *bufConn) error {
 	protocol := guessProtocol(sample)
 	switch protocol {
 	case "http/1":
+		return p.proxyHTTP1(cli, srv)
 	case "http/2":
+		// TODO: reintroduce HTTP/2 support
+		return p.proxyFallback(cli, srv)
 	default:
 		return p.proxyFallback(cli, srv)
```
Comment (sachin9996) [inline cmd/run/socket/proxy.go:249]: Were we just always falling back earlier? Not sure I understand the semantics here

File: journal/h1_test.go
Comment (sachin9996) [inline journal/h1_test.go:1]: Who needs tests anyway

**Review by sachin9996 — APPROVED** [review]

File: cmd/run/socket/proxy.go (hunk 1 above)
Comment (adtac) [reply cmd/run/socket/proxy.go:249]: No, I removed HTTP/2 support temporarily.

File: journal/h1_test.go
Comment (adtac) [reply journal/h1_test.go:1]: It's well-tested in the Go stdlib :)

**Pull request merged**